lib/molinillo/resolution.rb

```diff
@@ -689,7 +689,8 @@ def attempt_to_activate
       def attempt_to_filter_existing_spec(vertex)
         filtered_set = filtered_possibility_set(vertex)
         if !filtered_set.possibilities.empty? &&
-            (vertex.payload.dependencies == dependencies_for(possibility.latest_version))
+            (vertex.payload.dependencies == dependencies_for(possibility.latest_version) ||
+             vertex.payload.possibilities == possibility.possibilities)
```

Comment:
These lines are only here because `filtered_possibility_set` doesn't take into account the choice of possibility set (so the filtered set may be non-empty, but the new possibility is for a different possibility set, defined by its sub-dependencies, and should be rejected). At the time I wrote it, we had `search_for` not returning pre-release results unless the requirement was a pre-release. Since that's no-longer the case, I think a better fix here would be to rewrite that method:

```ruby
def filtered_possibility_set(vertex)
  PossibilitySet.new(vertex.payload.dependencies, vertex.payload.possibilities & possibility.possibilities)
end
```

With that done, we can just use `if !filtered_set.possibilities.empty?` here, without the dependencies check. That will fix this regression *and* speed things up 🎉 .